client/Client.cpp

```diff
@@ -761,6 +761,20 @@ void CClient::reinitScripting()
 #endif
 }
 
+void CClient::removeGUI()
+{
+	// CClient::endGame
+	GH.curInt = nullptr;
+	if(GH.topInt())
+		GH.topInt()->deactivate();
+	adventureInt = nullptr;
+	GH.listInt.clear();
+	GH.objsToBlit.clear();
+	GH.statusbar = nullptr;

```

Comment:
are curInt, adventureInt and statusBar shared_ptrs?
If so, please use reset() instead.
If not, isn't it a memory leak?

Reply:
This code was moved from different file without alteration

Reply:
Not a memory leak - `operator =` will correctly release object if needed.
In fact, shared_ptr does not have a way to release object, so no memory leaks are possible.
(unique_ptr does have release(), but not shared_ptr)

Reply:
BTW... curInt and LOCPLINT are... not very smart pointers :laughing: